### PR TITLE
sdk/node: support client cert authn

### PR DESCRIPTION
**Pull request by boymanjor**
The Client object can be passed an `https.Agent` with
its `ca`, `key`, and `cert` attributes set as file paths.

### CODE REVIEW

**Comment by boymanjor** [comment]
PTAL. Tested with https://gist.github.com/boymanjor/c6eae9847b7858c2aa9316b0ca35fa88

**Comment by jeffomatic** [comment]
Hot take: I think we're at the point that we should offer an alternative constructor that looks like this:

```
c = new Chain.Client({
  baseUrl: "https://...",
  token: ...,
  agent: ...
})
```

**Comment by boymanjor** [comment]
@jeffomatic I prefer that pattern of constructor function. Is that change a pre-requisite for landing this?

**Comment by jeffomatic** [comment]
@boymanjor Yeah, I feel bad about blocking here, but I'd rather not introduce a calling convention that we're going to deprecate right away.

Are you comfortable updating the existing constructor so it permissively accepts the old convention (one or two string params) and the new convention (a single object param)? If not I can sneak in a PR ahead of yours that introduces that.

**Comment by boymanjor** [comment]
@jeffomatic it's cool. I'll handle it.

**Comment by boymanjor** [comment]
@jeffomatic PTAL. Updated and tested with the following config:
```
const https = require('https')
const fs = require('fs')
const chain = require('chain-sdk')

let agent = new https.Agent({
	ca: fs.readFileSync(process.env.ROOT_CA_CERTS),
	cert: fs.readFileSync(process.env.TLSCRT),
	key: fs.readFileSync(process.env.TLSKEY)
})
const client = new chain.Client({
  baseUrl: 'https://localhost:1999',
  agent: agent
})
```

**Comment by jeffomatic** [comment]
LGTM but let's get @tonychain's sign-off here.

**Comment by boymanjor** [comment]
@jeffomatic totally.

File: sdk/node/src/connection.js [hunk 1]
```diff
@@ -66,11 +66,13 @@ class Connection {
...
    * @param {String} baseUrl Chain Core URL.
    * @param {String} token   Chain Core client token for API access.
+   * @param {String} agent   https.Agent used to provide TLS config.
    * @returns {Client}
    */
-  constructor(baseUrl, token = '') {
+  constructor(baseUrl, token = '', agent = {}) {
     this.baseUrl = baseUrl
     this.token = token || ''
+    this.agent = agent || {}
```
Comment (jeffomatic) [inline sdk/node/src/connection.js]: I think if you do `this.agent = agent` with no defaulting, you can avoid checking for object key count in the request function below.

**Comment by tonychain** [comment]
Haha, unfortunately my node-fu is not quite up to my Ruby. From what I can tell this seems fine though?

**Pull request merged**